zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template.rs

```diff
@@ -1,57 +1,96 @@
 //! The `GetBlockTempate` type is the output of the `getblocktemplate` RPC method.
 
-use crate::methods::get_block_template_rpcs::types::{
-    coinbase::Coinbase, default_roots::DefaultRoots, transaction::Transaction,
+use zebra_chain::{amount, block::ChainHistoryBlockTxAuthCommitmentHash};
+
+use crate::methods::{
+    get_block_template_rpcs::types::{
+        default_roots::DefaultRoots, transaction::TransactionTemplate,
+    },
+    GetBlockHash,
 };
 
 /// Documentation to be added after we document all the individual fields.
 #[derive(Clone, Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
 pub struct GetBlockTemplate {
     /// Add documentation.
     pub capabilities: Vec<String>,
-    /// Add documentation.
-    pub version: usize,
+
+    /// The version of the block format.
+    /// Always 4 for new Zcash blocks.
+    //
+    // TODO: add a default block version constant to zebra-chain.
+    pub version: u32,
+
     /// Add documentation.
     #[serde(rename = "previousblockhash")]
-    pub previous_block_hash: String,
+    pub previous_block_hash: GetBlockHash,
+
     /// Add documentation.
     #[serde(rename = "blockcommitmentshash")]
-    pub block_commitments_hash: String,
+    #[serde(with = "hex")]
+    pub block_commitments_hash: ChainHistoryBlockTxAuthCommitmentHash,
+
     /// Add documentation.
     #[serde(rename = "lightclientroothash")]
-    pub light_client_root_hash: String,
+    #[serde(with = "hex")]
+    pub light_client_root_hash: ChainHistoryBlockTxAuthCommitmentHash,
+
     /// Add documentation.
     #[serde(rename = "finalsaplingroothash")]
-    pub final_sapling_root_hash: String,
+    #[serde(with = "hex")]
+    pub final_sapling_root_hash: ChainHistoryBlockTxAuthCommitmentHash,
+
     /// Add documentation.
     #[serde(rename = "defaultroots")]
     pub default_roots: DefaultRoots,
-    /// Add documentation.
-    pub transactions: Vec<Transaction>,
-    /// Add documentation.
+
+    /// The non-coinbase transactions selected for this block template.
+    ///
+    /// TODO: select these transactions using ZIP-317 (#5473)
+    pub transactions: Vec<TransactionTemplate<amount::NonNegative>>,
+
+    /// The coinbase transactions generated from `transactions` and `height`.
```

Comment:
```suggestion
    /// The coinbase transactions generated for `transactions` and `height`.
```

Reply:
I think this is correct, we need transactions and height to generate the coinbase transaction?

But the plural is wrong.

```suggestion
    /// The coinbase transaction generated from `transactions` and `height`.
```